sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTransmitter.cs

```diff
@@ -101,42 +106,46 @@ private static TelemetryEnvelope GeneratePartAEnvelope(Activity activity)
 
         private MonitorBase GenerateTelemetryData(Activity activity)
         {
-            MonitorBase telemetry = new MonitorBase();
-
             var telemetryType = activity.GetTelemetryType();
-            telemetry.BaseType = Telemetry_Base_Type_Mapping[telemetryType];
-            string url = GetHttpUrl(activity.Tags);
+            var tags = activity.Tags.ToAzureMonitorTags(out var activityType);
```

Comment:
we should be using TagObjects instead of Tags. Tags only contain a subset of TagObjects which are string value.

Reply:
I have a TODO section, will get this handled in a separate PR. This changes requires a custom serialization to generated classes.